src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
-                _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
-                _setupFlowViewModel.StartAppManagementFlow(query);
-                await SearchAndSelectAsync(query);
+                await AppActivationFlowAsync(wingetIdsQuery, ActivationQueryType.WingetIds);
             }
         }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
-        }
     }
 
-    private async Task SearchAndSelectAsync(string query)
+    private async Task AppActivationFlowAsync(string query, ActivationQueryType queryType)
     {
-        var parameters = HttpUtility.ParseQueryString(query);
-        var searchParameter = parameters["search"];
+        if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+        {
+            _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
+            await _mainWindow.ShowErrorMessageDialogAsync(
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.AppInstallActivationTitle),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.URIActivationFailedBusy),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.Close));
+            return;
+        }
 
-        if (string.IsNullOrEmpty(searchParameter))
+        var identifiers = ParseIdentifiers(query, queryType);
+        if (identifiers.Length == 0)
         {
-            _log.Warning("Search parameter is missing or empty in the query.");
+            _log.Warning("No valid identifiers provided in the query.");
             return;
         }
 
-        // Currently using the first search term only
-        var firstSearchTerm = searchParameter.Split(Separator, StringSplitOptions.RemoveEmptyEntries)
-                                             .Select(term => term.Trim(' ', '"'))
-                                             .FirstOrDefault();
+        _log.Information("Starting add-apps-to-cart activation");
+        _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
+        _setupFlowViewModel.StartAppManagementFlow(queryType == ActivationQueryType.Search ? identifiers[0] : null);
+        await HandleAppSelectionAsync(identifiers, queryType);
+    }
 
-        if (string.IsNullOrEmpty(firstSearchTerm))
+    private async Task HandleAppSelectionAsync(string[] identifiers, ActivationQueryType queryType)
+    {
+        try
         {
-            _log.Warning("No valid search term was extracted from the query.");
-            return;
+            await SearchAndSelectAsync(identifiers);
         }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
+        }
+    }
+
+    private string[] ParseIdentifiers(string query, ActivationQueryType queryType)
+    {
+        switch (queryType)
```

Comment:
Now this, and `HandleInternalAsync` needs to stay in sync with the enumerator.  A lot of duplicated if/else.

Reply:
refactored with the change of input, so I'm resolving. I still have two switch statements, but I thought that was a better compromise vs having entirely separate methods for the two queryTypes given their overlap.